forc-plugins/forc-doc/src/render.rs

```diff
@@ -1231,3 +1231,46 @@ pub(crate) fn trim_fn_body(f: String) -> String {
         None => f,
     }
 }
+
+/// Checks if some raw html (rendered from markdown) contains a header.
+/// If it does, it splits at the header and returns the slice that preceeded it.
+pub(crate) fn split_at_markdown_header(raw_html: &str) -> &str {
+    const H1: &str = "<h1>";
+    const H2: &str = "<h2>";
+    const H3: &str = "<h3>";
+    const H4: &str = "<h4>";
+    const H5: &str = "<h5>";
+    if raw_html.contains(H1) {
+        let v: Vec<_> = raw_html.split(H1).collect();
+        match v.first() {
+            Some(s) => s,
+            None => panic!("expected some &str"),
+        }
```

Comment:
Rather than branching and `panic!`ing manually, we can use `expect`:
```suggestion
        v.first().expect("expected a non-empty str")
```
Same goes for the branches below.